server/utils/editAppointmentUtils.ts

```diff
@@ -431,3 +436,8 @@ export const hasAppointmentAttendeesChanged = (editAppointmentJourney: EditAppoi
 export function isUncancellable(appointment: AppointmentDetails): boolean {
   return appointment.isCancelled && parseDate(appointment.startDate) > subDays(startOfToday(), 6)
 }
+
+const isUncancelAndAllFutureNotCancelled = (editAppointmentJourney: EditAppointmentJourney) =>
+  editAppointmentJourney.uncancel &&
+  editAppointmentJourney.appointments.filter(i => toDate(i.startDate) >= startOfToday() && i.cancelled === false)
```

Comment:
I think you can use `some` instead of `filter` ?

Reply:
that's cleaner. I've changed it in the uncancel.ts file as well